### PR TITLE
Implement backfill for update posts records from COT

### LINKED ISSUE #32383: [COT] Implement backfill for shipping address into wp_posts and wp_postmeta table

### PR DESCRIPTION
**Pull request by vedanshujain**
### All Submissions:

* [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

This PR adds a method in `WC_Order_Data_Store_CPT` class called `update_order_from_object` which can update data into posts and postmeta tables from the object. We use this method to backfill data from custom table to wp_posts and wp_postmeta table.

Closes #32383, #32382, #32384, #32381.

### How to test the changes in this Pull Request:

Unfortunately, at this time there is no UI that this method feeds into, there at the moment it cannot be tested easily. See the unit test and check if it makes sense and covers basic use cases. We will likely update this method again after making more progress on the UX-related issues.

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [x] Have you written new tests for your changes, as applicable?
* [x] Have you successfully run tests with your changes locally?
* [x] Have you created a changelog file by running `pnpm nx affected --target=changelog`?

### FOR PR REVIEWER ONLY:

* [ ] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.

### CODE REVIEW

**Comment by vedanshujain** [comment]
Seems like tests are failing, checking

**Comment by botwoo** [comment]
:bar_chart: Test reports for this pull request have been published and are accessible through the following links:

- [REST API Test Report](https://woocommerce.github.io/woocommerce-test-reports/pr/32857/api/)
- E2E Test Report (TBC)

Latest commit referenced in the reports: `Applied coding standards.` 7ff0ba8061cde7c47d449cb073ebe86c386a7a3d
This comment will automatically be updated with the latest referenced commit when you push new changes to this pull request.

---

_Visit the [WooCommerce Test Reports](https://woocommerce.github.io/woocommerce-test-reports) homepage to view all published reports. See the [FAQs](https://github.com/woocommerce/woocommerce-test-reports/wiki/FAQs) page if you're having problems accessing them._

**Comment by botwoo** [comment]
:bar_chart: Test reports for 2ca4b42846a66d2878759ddbe189d9bc165fc523 `Modify tests to skip reading meta_ids` have been published and are accessible through the following links:

- [REST API Test Report](https://woocommerce.github.io/woocommerce-test-reports/pr/32857/api/)
- E2E Test Report (TBC)

Visit the [WooCommerce Test Reports](https://woocommerce.github.io/woocommerce-test-reports) homepage to view all published reports. See the [FAQs](https://github.com/woocommerce/woocommerce-test-reports/wiki/FAQs) page if you're having problems accessing them.

File: plugins/woocommerce/src/Internal/DataStores/Orders/OrdersTableDataStore.php [hunk 1]
```diff
@@ -427,13 +569,42 @@ public function read( &$order ) {
...
+	/**
+	 * Read metadata directly from database.
+	 *
+	 * @param \WC_Order $order Order object.
+	 *
+	 * @return array Metadata array.
+	 */
+	public function read_meta( &$order ) {
+		global $wpdb;
+		$meta_table = $this::get_meta_table_name();
```
Comment (jorgeatorres) [inline plugins/woocommerce/src/Internal/DataStores/Orders/OrdersTableDataStore.php:590]: I believe we should be using `self::get_meta_table_name()` to call class methods, but given this is implemented differently in #33026 I'm not going to worry too much about it.

**Review by jorgeatorres — APPROVED** [review]
Hi @vedanshujain!

Thanks for this PR. I tested through the `add/32667-cot-data-store-update` in order to be able to read/update metadata correctly and manually running code similar to the one in the tests, I see that the posts are updated accordingly.

I didn't test every possible combination, but I can try to do that in #33026. Also, merging this would simplify #33026 a little, which is useful.

LGTM 👍.

**Pull request merged**